src/System.Private.Xml/src/System/Xml/XmlDownloadManager.cs

```diff
@@ -33,6 +34,27 @@ internal partial class XmlDownloadManager
 
         private Stream GetNonFileStream(Uri uri, ICredentials credentials, IWebProxy proxy,
             RequestCachePolicy cachePolicy)
+        {
+            WebRequest req = CreateWebRequestOrThrowIfRemoved(uri, credentials, proxy, cachePolicy);
+
+            using (WebResponse resp = req.GetResponse())
+            using (Stream respStream = resp.GetResponseStream())
+            {
+                var result = new MemoryStream();
+                respStream.CopyTo(result);
+                result.Position = 0;
+                return result;
+            }
+        }
+
+        // This method is marked Removable because WebRequest has a lot of dependencies that will bloat
+        // self-contained distributions of .NET Apps.
+        // This code is statically reachable from any place that uses XmlReaderSettings (i.e. every app that
+        // does something XML related is going to have this in their transitive call graph). People rarely need
+        // this functionality though.
+        [RemovableFeature("System.Xml.XmlUrlResolver.NonFileUrlSupport")] 
+        private static WebRequest CreateWebRequestOrThrowIfRemoved(Uri uri, ICredentials credentials, IWebProxy proxy,
```

Comment:
Does it make a difference if you put this attribute directly on GetNonFileStream?

Reply:
It was just convenient to have a single choke point that works for both the async and synchronous versions of this API.